src/Datadog.Trace/AspNetCore/DatadogLoggingScope.cs

```diff
@@ -0,0 +1,101 @@
+// <copyright file="DatadogLoggingScope.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Datadog.Trace.AspNetCore
+{
+    // Implementation based on AspNetCore Hosting implementation: https://github.com/dotnet/aspnetcore/blob/1a2b3260c6161ae9b7f639de228a6eb0488a1075/src/Hosting/Hosting/src/Internal/HostingLoggerExtensions.cs#L114
+    internal class DatadogLoggingScope : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private readonly string _service;
+        private readonly string _env;
+        private readonly string _version;
+        private readonly string _traceId;
+        private readonly int _count;
+
+        private string _cachedToString;
+
+        public DatadogLoggingScope()
+            : this(Tracer.Instance)
+        {
+        }
+
+        internal DatadogLoggingScope(Tracer tracer)
+        {
+            _service = tracer.DefaultServiceName ?? string.Empty;
+            _env = tracer.Settings.Environment ?? string.Empty;
+            _version = tracer.Settings.ServiceVersion ?? string.Empty;
+            var traceId = tracer.ActiveScope?.Span?.TraceId.ToString();
+            _traceId = traceId ?? "0";
+            _count = traceId is null ? 3 : 4;
+        }
+
+        public int Count
+        {
+            get
+            {
+                return _count;
+            }
+        }
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (index == 0)
+                {
+                    return new KeyValuePair<string, object>("dd_service", _service);
```

Comment:
These identifiers are "serilog" style. If users are using ILogger injection _and_ NLog/Log4NET, then they will have two sets of identifiers I think. Not sure what the best way to tackle that is

Reply:
Yeah they'll have two sets of identifiers...perhaps we'll have a better idea of how to tackle it once we see it in use?